File: Data Structures & Algorithms/max-consecutive-ones/submission-2.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int m = 0;
        int a = 0;
        for (int i = 0; i < nums.length; i++){
            if (nums[i] == 0) { a = 0;}
            else a += 1;
            if (a > m) {m = a;}
        }
        return m;

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        output = [1]*len(nums)

        l = 1

        for i in range(len(nums)):
            output[i] = l
            l *= nums[i]

        r = 1

        for i in range(len(nums)-1,-1,-1):
            output[i]=r *output[i]
            r = nums[i]*r

        return output
